source/common/event/dispatcher_impl.cc

```diff
@@ -119,9 +119,9 @@ DispatcherImpl::createSslListener(Network::ConnectionHandler& conn_handler,
 }
 
 TimerPtr DispatcherImpl::createTimer(TimerCb cb) {
-  // TODO(htuch): Figure out why we can't set this assertion, see
-  // https://github.com/lyft/envoy/issues/1294.
-  // ASSERT(isThreadSafe());
+  // We don't ASSERT that createTimer() is called on the run thread because timers are used for
```

Comment:
It makes sense that some timer ops might be cross thread (maybe?) but not that the timer creation is cross-thread. From what you write after, it seems safe, but is there some anti-pattern involved here?